WalletWasabi/WabiSabi/Backend/Rounds/Arena.cs

```diff
@@ -214,6 +217,11 @@ private async Task StepConnectionConfirmationPhaseAsync(CancellationToken cancel
 						var offendingAliceCounter = 0;
 						await foreach (var offendingAlices in CheckTxoSpendStatusAsync(round, cancel).ConfigureAwait(false))
 						{
+							if (cancel.IsCancellationRequested)
+							{
+								break;
+							}
```

Comment:
Why not
```suggestion
							cancel.ThrowIfCancellationRequested();
```

Throwing here would end the round properly + log.